packages/astro/test/fixtures/middleware space/src/middleware.js

```diff
@@ -24,7 +24,7 @@ const first = defineMiddleware(async (context, next) => {
 		const response = await next();
 		const newResponse = response.clone();
 		const /** @type {string} */ html = await newResponse.text();
-		const newhtml = html.replace('<h1>testing</h1>', '<h1>it works</h1>');
+		const newhtml = html.replace('testing', 'it works');
```

Comment:
So, this is a bit awkward. But since we now add attributes, those kind of naive replacements don't work in dev.

In my humble, informed, opinion, you shouldn't rely on string replaces like this anyway, because whitespaces, minification etc etc could ruin it anyway. Typically, I'd recommend people to use a real tool for this, like https://github.com/cloudflare/lol-html

Reply:
wow, such shade 🌨